### ISSUE TITLE
Make it compatible with a Wayland compositor

### ISSUE THREAD

**Issue by matteodelabre** [matteodelabre/waved#9]

**Comment by alistair23** [comment]
Hey! I wanted to check what the status of this is?

**Comment by matteodelabre** [comment]
Hi @alistair23! It’s high on my todo list, but I didn’t get time to work on waved recently. My plan is first to improve the performance of the frame generation loop (https://github.com/matteodelabre/waved/blob/a28654d42ed8646e8536b03a353c7b52d65aed9e/src/display.cpp#L494-L571) and then to then implement #4 along with this issue. I’ll make sure to update this issue when I get started on the implementation!

**Comment by tadfisher** [comment]
Would it not be a better option (in the long run) to get the `mxsfb-drm` driver running? That would give us a more modern driver to build upon, and more options for Wayland compositors (the popular ones don't have fbdev backends).

**Comment by alistair23** [comment]
I have support in the kernel, see https://github.com/alistair23/linux/commit/6f967006081e2b2b450a1eae1cfda988aa216701

I think in the long run that would be a better option. It's probably not a bad idea to get waved working first with fbdev and then work on drm support after that.

**Comment by matteodelabre** [comment]
I may be missing something, but I can’t see the advantage in using `mxsfb-drm`. To clarify, when waved is fully operational, the stack will be as follows:

```
EPDC <> mxsfb <> waved <> Compositor <> User apps
```

So changing the underlying driver will only impact the interface that waved deals with, and won’t affect the Wayland compositors as far as I can see.

**Comment by alistair23** [comment]
I agree, I don't think it'll affect the compositor. It will make it easier to run a mainline kernel though as the current fbdev emulation isn't good enough to run Xoichtl
